routes/api/v1/signup.js

```diff
@@ -0,0 +1,29 @@
+var express = require('express');
+var router = express.Router();
+
+router.post('/', (req, res, next) => {
+  var db = req.app.get('db')
+  var user = req.body
+
+  db('users').where({username: user.username})
+  .then( (existingUser) => {
+    if(existingUser[0] !== undefined) {
+      return res.json({errorMessage: 'Username already exists'})
+    } else {
+      if(user.username !== undefined && user.password !== undefined) {
+        //hash password with bcrypt before saving
+        db('users').insert(user)
+        .then( () => {
+          //return JWT
+          return res.json()
+        });
+      } else {
+        var err = new Error('Not found');
+        err.status = 400;
+        next(err);
+      }
+    }
+  });
```

Comment:
@wilmoore 
This is my post for creating a user. I have written three tests (see below) and they are passing. My next thought would be to refactor. What are your thoughts on where this logic should live?